polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp

```diff
@@ -44,6 +45,50 @@ using namespace mlir::polygeist;
 
 namespace {
 
+//===----------------------------------------------------------------------===//
+// WorkGroupSize
+//===----------------------------------------------------------------------===//
+
+/// Class to represent work group size of a kernel.
+class WorkGroupSize {
+public:
+  using ElemTy = std::variant<Value, unsigned>;
+  WorkGroupSize(Value ndItem, const sycl::ReqdWorkGroupSize &reqdWorkGroupSize,
+                OpBuilder builder) {
+    auto ndItemTy = cast<sycl::NdItemType>(
+        cast<MemRefType>(ndItem.getType()).getElementType());
+    for (unsigned dim = 0; dim < ndItemTy.getDimension(); ++dim) {
+      /// Use constants provided by reqd_work_group_size if given (non-empty).
+      if (!reqdWorkGroupSize.empty())
+        wgSizes.push_back(reqdWorkGroupSize[dim]);
```

Comment:
`reqdWorkGroupSize` **should** have the same number  of dimensions as `ndItemTy`. Add an assert to ensure it is the case.